traits-stubs/traits-stubs/trait_types.pyi

```diff
@@ -0,0 +1,426 @@
+    ...
+
+
+# -----------------BaseCFloat--------------------
+
+class BaseCFloat(BaseFloat):
+    ...
+
+
+class CFloat(BaseCFloat):
+    ...
+
+
+# -----------------BaseCComplex--------------------
+
+class BaseCComplex(BaseComplex):
+    ...
+
+
+class CComplex(BaseCComplex):
+    ...
+
+
+# -----------------BaseCStr--------------------
+
+class BaseCStr(BaseStr):
+    ...
+
+
+class CStr(BaseCStr):
+    ...
+
+
+# -----------------BaseCBytes--------------------
+
+class BaseCBytes(BaseBytes):
+    ...
+
+
+class CBytes(BaseCBytes):
+    ...
+
+
+# -----------------BaseCBool--------------------
+
+class BaseCBool(BaseBool):
+    ...
+
+
+class CBool(BaseCBool):
+    ...
+
+
+# -----------------String--------------------
+
+class _String(_TraitType[_T, str]):
+    ...
+
+
+class String(_String[str]):
+    def __init__(self, value: str = ..., minlen: int = ..., maxlen: int = ...,
+                 regex: str = ...):
+        ...
+
+
+# -----------------Regex--------------------
+
+class Regex(String):
+    ...
+
+
+# -----------------Code--------------------
+
+class Code(String):
+    ...
+
+
+# -----------------HTML--------------------
+
+class HTML(String):
+    ...
+
+
+# -----------------Password--------------------
+
+class Password(String):
+    ...
+
+
+# -----------------Callable--------------------
+
+class _BaseCallable(_TraitType[_T, _CallableType[..., _Any]]):
+    ...
+
+
+class BaseCallable(_BaseCallable[_CallableType[..., _Any]]):
+    ...
+
+
+class Callable(BaseCallable):
+    ...
+
+
+# -----------------Type--------------------
+
+# TODO
+class BaseType(_Any):
+    ...
+
+
+class Type(BaseType):
+    ...
+
+
+# -----------------This--------------------
+class This(BaseType):
+    ...
+
+
+# -----------------self--------------------
+class self(This):
+    ...
+
+
+# -----------------Function--------------------
+class Function(_TraitType):
+    ...
+
+
+# -----------------Method--------------------
+
+class Method(_TraitType):
+    ...
+
+
+# -----------------Module--------------------
+
+class Module(_TraitType):
+    ...
+
+
+# -----------------Python--------------------
+class Python(_TraitType):
+    ...
+
+
+# -----------------ReadOnly--------------------
+# TODO
+class ReadOnly(_Any):
+    ...
+
+
+# -----------------Disallow--------------------
+# TODO
+class Disallow(_Any):
+    ...
+
+
+# -----------------Constant--------------------
+class Constant(_TraitType):
+    ...
+
+
+# -----------------Delegate--------------------
+class Delegate(_TraitType):
+    ...
+
+
+# -----------------DelegatesTo--------------------
+class DelegaatesTo(Delegate):
+    ...
+
+
+# -----------------PrototypedFrom--------------------
+class PrototypedFrom(Delegate):
+    ...
+
+
+# -----------------Expression--------------------
+class Expression(_TraitType):
+    ...
+
+
+# ---------------------PythonValue----------------
+class PythonValue(_Any):
+    ...
+
+
+# ----------------BaseFile---------------------
+class _BaseFile(_BaseStr):
```

Comment:
This also accepts `pathlib.Path` instances (there may be an abstract base class that we want to use instead).